usr/lib/pkcs11/common/object.c

```diff
@@ -144,28 +140,48 @@ CK_RV object_create(STDLL_TokData_t * tokdata,
             goto error;
         }
 
-        flag = *(CK_BBOOL *) attr->pValue;
-        flag = (~flag) & 0x1;
-
+        flag = CK_FALSE;
         rc = build_attribute(CKA_NEVER_EXTRACTABLE, &flag, sizeof(CK_BYTE),
                              &extractable);
         if (rc != CKR_OK) {
             TRACE_DEVEL("build attribute failed.\n");
             goto error;
         }
+
         template_update_attribute(o->template, sensitive);
         template_update_attribute(o->template, extractable);
+        /* fall-through */
```

Comment:
Hmm, so we require that CKA_LOCAL is already there, otherwise we fail.
If it is there, we ignore its value, and set it to FALSE in any case....
Sounds somewhat strange to me. Can't we just do template_update_attribute with the attribute set to FALSE in any case?

Function key_object_set_default_attributes (in key.c) already adds and sets CKA_LOCAL to FALSE. This is called from template_add_default_attributes for all key types inside object_create_skel which is called just above. So the code you added might not be needed at all.

Reply:
Have to look into the CKA_LOCAL thing ie if its already set to false here. Some attribute set clearly wrong here leading to test case failure , see my mail.

Problem was because the comment above (that i removed) said:
> The PKCS #11 spec doesn't impose this restriction however.

... but it does, at least for private and secret keys.

Reply:
CKA_LOCAL is already present and set to false, as you expected. fixed that